statsmodels/tsa/automatic/tests/results/autotransform.R

```diff
@@ -0,0 +1,3 @@
+library("forecast")
+macrodata <- read.csv('/Users/abhijeet/gsoc/dev/statsmodels/statsmodels/datasets/macrodata/macrodata.csv')
+lambda <- BoxCox.lambda(macrodata['cpi'], method = 'loglik', lower = -1, upper = 2)
```

Comment:
Here you are passing BoxCox.lambda a `data.frame` object, which apparently it does not know how to handle and always returns `1`. Instead, use the following test cases:

```R
# Here y is not a ts() object, so only is fit with a mean
y <- macrodata$cpi
lambda_d0s1 <- BoxCox.lambda(y, method = 'loglik', lower = -1, upper = 2)

# y is now a ts() object, so it now includes a time trend, but no seasonal
y <- ts(macrodata$cpi)
lambda_d1s1 <- BoxCox.lambda(y, method = 'loglik', lower = -1, upper = 2)

# y is now a ts() object w/ frequency, so it now includes a time trend
# and a seasonal
y <- ts(macrodata$cpi, frequency=12)
lambda_d1s12 <- BoxCox.lambda(y, method = 'loglik', lower = -1,
                                       upper = 2)
```

I've indicated the appropriate test variables in the names of the return variables, so e.g. `lambda_d0s1` should correspond to your test case where `d = 0` and `seasonal_periods=1`.

Reply:
@ChadFulton Thank you for pointing out my errors, I have the necessary changes to the unit test and now it's passing. Thank you for helping me. I'll add a few more unit tests.